File: LearnCSharpTests/Console/ProgramTests.cs
using System;
using System.IO;
using LearnCSharpLibrary;
using Xunit;

namespace LearnCSharpTests
{
    public class ProgramTests
    {
        [Fact]
        public void MainTest()
        {
            var consoleOutput = new StringWriter();
            Console.SetOut(consoleOutput);

            var bootstrapTest = new Bootstrap(new[] { "MainTest" });
            bootstrapTest.Start();
            var expected = $"Hello World from MainTest!{Environment.NewLine}";
            Assert.Equal(expected, consoleOutput.ToString());
        }
    }
}

File: LearnFSharpTests/Executable/ProgramTests.fs
module LearnFSharpTests.Console.ProgramTests

open System
open System.IO
open Xunit
open LearnFSharpLibrary

[<Fact>]
let MainTest() =
    let consoleOutput = new StringWriter()
    Console.SetOut consoleOutput
    Bootstrap.start [| "MainTest" |] |> ignore
    let expected = sprintf "Hello World from MainTest!%s" Environment.NewLine
    Assert.Equal(expected, consoleOutput.ToString())
